test/cpp/end2end/xds/xds_end2end_test.cc

```diff
@@ -321,7 +322,20 @@ class XdsSecurityTest : public XdsEnd2endTest {
       const std::vector<StringMatcher>& san_matchers,
       const std::vector<std::string>& expected_authenticated_identity,
       bool test_expects_failure = false) {
+    // Change the backend and use a unique service name to use so that we know
+    // that the CDS update was applied.
+    std::string service_name = absl::StrCat(
+        "eds_service_name",
+        absl::FormatTime("%H%M%E3S", absl::Now(), absl::LocalTimeZone()));
```

Comment:
Instead of dynamically creating a new EDS resource each time and using a timestamp in the name, I suggest changing `SetUp()` to unconditionally create a second EDS resource pointing to backend 1 with a fixed name, and then toggling between the two EDS names here.  I'm thinking you can use something like this:

```
backend_index_ = (backend_index_ + 1) % 2;
std::string eds_resource_name =
    absl::StrCat(kDefaultEdsServiceName, backend_index_ > 0 ? backend_index_ : "");
```

Reply:
That limits the number of updates we can do within a test to two. If we perform more updates, we'll end up using a previously connected subchannels. I can ofcourse add more backends, but I suspect that we won't realize this till we end up adding more updates in a single test and end up with flaky tests. Generating dynamic resource names sounds like a better idea to me.

Reply:
I am surprised that is an issue, since we should unref the old subchannel once we switch to the new one, and if we switch back to the old one, we should wind up recreating it.

But I guess this isn't worth spending time on right now.  We can reevaluate this when we eventually come back to restructure these tests as a whole.